### PR TITLE
Added code to import recent dates first and future dates at last

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added code to import recent dates first and future dates at last
Fixes: DEV-2984

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by tsteur — COMMENTED** [review]
I haven't tested it but looks good. Is it maybe possible to add a test for the new private method just to be safe it works?

**Comment by AltamashShaikh** [comment]
@tsteur 
Added testcase for above method - https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/055c8a3ed00a5ca91721a8d89cc03889edbf6afb/tests/Integration/ImporterTest.php#L61

Please check

**Review by tsteur — COMMENTED** [review]
Nice thanks! Can we also add one more test case for the start and end date being in the past? Just to be safe?

**Comment by AltamashShaikh** [comment]
> Nice thanks! Can we also add one more test case for the start and end date being in the past? Just to be safe?

@tsteur Added, please check

File: templates/index.twig [hunk 1]
```diff
@@ -9,7 +9,7 @@
         {{ 'GoogleAnalyticsImporter_ImporterHelp3'|translate('<br><br><strong>', '</strong>')|raw }}
     </div>
     <div piwik-content-block
-         content-title="{{ title|e('html_attr') }}"
+         content-title="{{ title }}"
```
Comment (tsteur) [inline templates/index.twig:12]: @AltamashShaikh is this change on purpose?

**Review by tsteur — COMMENTED** [review]
Nice @AltamashShaikh looks generally good and ready to merge. I've left one more comment where an escaping was removed that was maybe by accident? Or can we be sure there won't be any XSS possible?

File: templates/index.twig (hunk 1 above)
Comment (AltamashShaikh) [reply templates/index.twig:12]: @tsteur yes, recently a customer reached out to us with 1 issue in GAImporter which was due to vue files not loading in wordpress correctly and we noticed this issue for french users
dc6b6289-f1b1-4b6a-8c9f-a70ac41a506e

If you see "Configure the Importer" text, its getting escaped due `'`

Comment (AltamashShaikh) [reply templates/index.twig:12]: since we were fixing this issue and was a very small one I included in this only

Comment (tsteur) [reply templates/index.twig:12]: OK sounds good as long as there's no user input 👍

Comment (AltamashShaikh) [reply templates/index.twig:12]: Yes, its just the text from translation

Comment (tsteur) [reply templates/index.twig:12]: feel free to merge 🎉
